### PR TITLE
[JS OV] Add Core.importModel()

### PR DESCRIPTION
**Pull request by almilosz**
### Details:
 - Add natively asynchronous Core.importModel()
 - Add definition to addon.ts

### Tickets:
 - *136462*

### CODE REVIEW

**Review by vishniakov-nikolai — COMMENTED** [review]
Main comment that each async method has the same structure, but looks different. 
For now it works but it's hard to read and support.
Could you propose solution that implements async behavior, that extracted to separated file and well documented.

File: src/bindings/js/node/src/core_wrap.cpp [hunk 1]
```diff
@@ -13,6 +13,10 @@
 #include "node/include/type_validation.hpp"
 #include "openvino/util/common_util.hpp"
 
+namespace {
+std::mutex core_mutex;
+}
```
[the remark this answers is not in the document]
Comment (almilosz) [reply src/bindings/js/node/src/core_wrap.cpp]: It can be a class member

**Review by vishniakov-nikolai — APPROVED** [review]
JS bindings contains 3 async methods:
- `read_model_async`
- `compile_model_async`
- `infer_async`

Each of them uses own structure:
image
image
image

You are adding fourth. 
I propose to extract common part from structures in parent class/structure, to specify it in each async method to do not duplicate code.

Approved for now, but tech debt increased.

**Comment by almilosz** [comment]
Agree that the approach to asynchronous operations should be unified. I created a [ticket](https://jira.devtools.intel.com/browse/CVS-146325) containing a comment regarding TsfnContexts structs. 
Those changes will affect a few parts of the code

**Pull request merged**